Isolate calico files

djosborne opened the pull request.

djosborne commented:
@karya0 does `net-modules.spec` need updating?

ConnorDoyle commented on .gitignore:
```diff
@@ -1,4 +1,3 @@
-calico-node-*.tar
 # First, ignore yourself.
 .gitignore
```
Not part of your changes, but why is this here?

karya0 commented:
I have a high-level question: why move `isolator` to `isolator/isolator`? I was imagining more like a top-level `calico` directory and calico specific bits in there without changing anything related to `isolator`.

ConnorDoyle commented on .gitignore:
```diff
@@ -27,6 +26,10 @@ missing
+docker-compose
+
 # OSX Resource
 .DS_Store
```
Again, not necessarily related to this change but this stuff IMO belongs in a user's global configuration.  No reason to have OS-specific targets here.  Mind cleaning this up?

ConnorDoyle commented on Dockerfile:
```diff
@@ -89,3 +59,26 @@ ADD ./init_scripts/etc/service/docker/run /etc/service/docker/run
+# Isolator
+####################
+WORKDIR /isolator
+ADD ./isolator/ /isolator/
```
:+1:

djosborne commented:
@karya0 A lot of the isolator specific code was scattered around the project's root directory, and moved around during the docker build. I moved it into an isolator folder so that the docker build steps could be reduced from
```
ADD ./isolator /isolator/isolator/
ADD ./m4 /isolator/m4/
ADD ./bootstrap /isolator/
ADD ./calico_isolator /isolator/
ADD ./configure.ac /isolator/
ADD ./Makefile.am /isolator/
``` 

to :
```
ADD ./isolator/ /isolator/
```

ConnorDoyle commented on Makefile:
```diff
@@ -10,14 +10,14 @@ docker-compose:
-		docker rm -f $$container; true; \
-	done
+	docker-compose kill
+	docker-compose -p netmodules  rm --force
```
nit: extra space?

ConnorDoyle commented on docker-compose.yml:
```diff
@@ -61,11 +34,10 @@ slave2:
     - MESOS_ISOLATION=com_mesosphere_mesos_NetworkIsolator
     - MESOS_HOOKS=com_mesosphere_mesos_NetworkHook
     - ETCD_AUTHORITY=etcd:4001
-    - MESOS_ATTRIBUTES=public:true
```
I think this will break the demo, since we target the "public" slave to launch the visualizer.

djosborne replied:
The demo is working if you hit `192.168.255.253:9001` directly. What is responsible for enforcing this policy on non-public slaves? Should the framework recognize that its on a non-public slave and communicate that to calico?

ConnorDoyle replied:
We tagged one as public in order to know deterministically which slave that task will land on since we created a route to that slave from outside the VM (via `add-container-route.sh`).  It looks like someone has since deleted that constraint from `stars.json` and `stars-isolated.json`.  Now it's up to chance that it continues to be scheduled on slave1.  I am in favor of re-introducing the attribute and the constraint to reduce the magic (relying on available resources and Marathon's current offer matching logic).

djosborne replied:
@ConnorDoyle 
There doesn't actually need to be a requirement that the nexthop route added via `add-container-route.sh` belongs to the Slave which the UI is running on. Since each Slave is also BGP speaker, even if the app ends up on Slave2, Slave1 will be able to nexthop the HTTP request to Slave2.

I've confirmed this by swapping the route added by `add-container-route.sh` to the other Slave and the demo still loads fine.

ConnorDoyle replied:
Awesome, thanks for the explanation.

spikecurtis replied:
I think we might have used it for the "before-Calico" demo, so that we could statically port map from outside the VM.  But, we stopped showing the collector in that part of the live demo.

ConnorDoyle commented on docker-compose.yml:
```diff
@@ -18,36 +18,9 @@ mesosmaster:
   build: .
   privileged: true
-  ports:
-    - "5051:5051"
```
Is it still possible to follow the links in the Mesos UI to the slave sandbox without this mapping?

djosborne replied:
Yes, sandbox links are all still working as expected without this. Can you think of any other reasons this port mapping exists? I'm not familiar with why its here. But you can't scale a service  in docker-compose if you have a port mapping set, which is why I removed it

ConnorDoyle replied:
Nope that's the only reason I can think of right now.

ConnorDoyle commented:
@djosborne the simplifications you made in `docker-compose.yml` are welcome, I'm just concerned they may compromise the demo.  Before I try to fire it up, were you able to verify that these changes didn't break anything?

djosborne commented:
@ConnorDoyle I will confirm that the demo still works. (I was working on too many branches at once, I don't  think I reran it before I submitted this PR)

spikecurtis commented:
LGTM!

ConnorDoyle commented:
Yep, :shipit:
